src/README.md

```diff
@@ -0,0 +1,464 @@
+Request received for profile 0x01
+response: 0,1,1597
+{"output":"0x00000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000001000000000000000000000000000000000000000000000000000000000000063d"}
+➜  example
+```
+
+We have now successfully tested the default function and ran a test to verify the function returns a response as expected.
+
+### Testing Default Function with Local Hardhat Node
+
+Previously we showed how to test the default function locally without a running node, but we can also run two other tests.
+
+1. Run local hardhat node and run the default mocha e2e tests.
+2. Run local hardhat node and watch the requests that are pushed and see how the function transforms the data.
+
+**Run local hardhat node and run the default mocha e2e tests**
+
+Lets’s start with the first test case. First we will start a local hardhat node.
+
+> Note: You will need to ensure you configure your local vars `POLYGON_RPC_URL` and `MUMBAI_RPC_URL` `.env` file. You can do this with `cp .env.local .env` then edit the `.env` with your information.
+>
+<details>
+  <summary>Expected error if <code>.env</code> not configured.</summary>
+
+    ➜  example yarn hardhat node                                                                                                       ~/Projects/Phala/example
+    yarn run v1.22.18
+    warning ../../../package.json: No license field
+    $ /Users/hashwarlock/Projects/Phala/example/node_modules/.bin/hardhat node
+    Error HH8: There's one or more errors in your config file:
+    
+      * Invalid value undefined for HardhatConfig.networks.polygon.url - Expected a value of type string.
+      * Invalid value undefined for HardhatConfig.networks.mumbai.url - Expected a value of type string.
+      
+    To learn more about Hardhat's configuration, please go to https://hardhat.org/config/
+    
+    For more info go to https://hardhat.org/HH8 or run Hardhat with --show-stack-traces
+    error Command failed with exit code 1.
+    info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
+</details>
+
+```bash
+➜  example yarn hardhat node                                                                                                       ~/Projects/Phala/example
+yarn run v1.22.18
+warning ../../../package.json: No license field
+$ /Users/hashwarlock/Projects/Phala/example/node_modules/.bin/hardhat node
+Started HTTP and WebSocket JSON-RPC server at http://127.0.0.1:8545/
+
+Accounts
+========
+
+WARNING: These accounts, and their private keys, are publicly known.
+Any funds sent to them on Mainnet or any other live network WILL BE LOST.
+```
+
+With our hardhat node running locally, we can now execute the e2e test.
+
+```bash
+➜  example yarn hardhat test --network localhost                                                                                   ~/Projects/Phala/example
```

Comment:
No need to start a local hardhat node, just run
```shell
yarn hardhat test
```
If no network is specified, the test runs a node by itself.